### ISSUE TITLE
There is false positive report for E0061

### ISSUE THREAD

**Issue by mili-l** [intellij-rust/intellij-rust#8823]
## Environment

* **IntelliJ Rust plugin version:** 0.4.171.4611-221-nightly
* **Rust toolchain version:** 1.62.0-nightly (60e50fc1c 2022-04-04) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.1 (CL-221.5080.224)
* **Operating system:** macOS 11.5.1
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
There is false positive report for E0061

## Steps to reproduce
```rust
use std::ops::Deref;

fn main() {
    dbg!(Test("abcdefg"));  // <-- here 
}

struct Test;

impl Deref for Test {
    type Target = fn(&str) -> &str;

    fn deref(&self) -> &Self::Target {
        &(mapper as _)
    }
}

fn mapper(text: &str) -> &str {
    text
}
```

## Actual Results

<img width="643" alt="Screenshot 2022-05-10 at 10 32 05" src="https://user-images.githubusercontent.com/1213122/167585280-8a5dbdbc-95d5-4373-9be3-7c2e11557bb8.png">

Related to #8790

**Comment by missingdays** [comment]
This issue doesn't reproduce on the latest plugin release so I'm closing it
Please reopen this issue if it still reproduces for you
